### ISSUE TITLE
Jodit type errors when running project with this package

### ISSUE THREAD

**Issue by ScottMalmberg** [Fedeorlandau/jodit-react-ts#1]
Hey there! I found your project through an issue on `jodit-react`, but I'm running into a lot of TS errors when starting up my project with it. It doesn't happen using `jodit-react`, but after installing this one I'm getting TS errors inside of the `jodit` `node_module` file. Things like:

```
ERROR in /Users/scottmalmberg/Dev/jodit-react-ts/node_modules/jodit/src/modules/table.ts
741:31-36
[tsl] ERROR in /Users/scottmalmberg/Dev/jodit-react-ts/node_modules/jodit/src/modules/table.ts(741,32)
      TS6133: 'index' is declared but its value is never read.

ERROR in /Users/scottmalmberg/Dev/jodit-react-ts/node_modules/jodit/src/jodit.ts
684:39-43
[tsl] ERROR in /Users/scottmalmberg/Dev/jodit-react-ts/node_modules/jodit/src/jodit.ts(684,40)
      TS6133: 'type' is declared but its value is never read.
```

I've tried removing my `yarn.lock` file and updating my TypeScript (v4.2.2) and ts-loader (v8.0.14) versions to match Jodit's, but haven't had luck. Initially the install seemed successful when I installed while my webpack-dev-server (v3.11.2) was running, but the errors popped up after I restarted it.

Any idea what could be wrong? Thanks!

**Comment by Fedeorlandau** [comment]
Hi @ScottMalmberg ! The errors come from Jodit itself. Their [types](https://github.com/xdan/jodit/issues/619) aren't the best. 

For typescript projects you should add https://www.typescriptlang.org/tsconfig/skipLibCheck.html to your tsconfig.json file.

This will build your project regardless any TS error on node_modules/. Read more [here](https://www.reddit.com/r/typescript/comments/aa3ui7/question_why_skiplibcheck_true_is_not_the_default/)

```
{
 "skipLibCheck": true
}
```

If this doesn't work please let me know,

Cheers

**Comment by ScottMalmberg** [comment]
Thanks for the response! Unfortunately, that flag is already set in my project. Strange that no errors come through when using `jodit-react`, too. I am able to run your example after cloning into this repo, but not when I try to add it to any of my existing projects.

**Comment by Fedeorlandau** [comment]
Could you please share me your tsconfig.json? It would be super helpful for Jodit and I. What kind of framework are you using? Nextjs, gatsby? Create-react-app? Cheers,

**Comment by ScottMalmberg** [comment]
Definitely! It's a React project with Webpack in a mono-repo. This is the `tsconfig` in the specific project I'm trying to use it in:
> {
    "compilerOptions": {
      "allowUnreachableCode": false,
      "allowUnusedLabels": false,
      "allowJs": false,
      "allowSyntheticDefaultImports": false,
      "alwaysStrict": true,
      "experimentalDecorators": true,
      "forceConsistentCasingInFileNames": true,
      "sourceMap": true,
      "inlineSourceMap": false,
      "jsx": "react",
      "module": "commonjs",
      "moduleResolution": "node",
      "noEmitOnError": false,
      "noFallthroughCasesInSwitch": true,
      "noUnusedLocals": true,
      "noUnusedParameters": true,
      "noImplicitReturns": true,
      "noImplicitThis": true,
      "preserveConstEnums": false,
      "pretty": true,
      "outDir": "./lib",
      "rootDir": "./src",
      "strictNullChecks": true,
      "suppressExcessPropertyErrors": false,
      "target": "es6",
      "skipLibCheck": true,
      "typeRoots": [
        "./node_modules/@types",
        "../../node_modules/@types",
        "./typings"
      ],
    },
    "include": [
      "./src/**/*"
    ],
    "exclude": [
      "node_modules",
      "lib"
    ]
  }

And this is the `tsconfig` in the root of my mono-repo:

> {
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "sourceMap": true,
    "noImplicitAny": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "declaration": true,
    "lib": [
      "es5",
      "es6",
      "dom"
    ],
    "jsx": "react",
    "types": [
      "node"
    ],
    "sourceRoot": "src",
    "declarationDir": "dist",
    "outDir": "dist",
    "typeRoots": [
      "./node_modules/@types"
    ]
  },
  "exclude": [
    "node_modules"
  ]
}

**Comment by mhammadsiddiqui** [comment]
I am having th e same issue while using,
"jodit-react": "^1.1.1",
"typescript": "^4.4.2"
and "skipLibCheck": true, option in my tsconfig.json.

I can not do gulp serve. Is it due to tslint ?

**Comment by Fedeorlandau** [comment]
> I am having th e same issue while using,
> 
> "jodit-react": "^1.1.1",
> 
> "typescript": "^4.4.2"
> 
> and "skipLibCheck": true, option in my tsconfig.json.
> 
> 
> 
> I can not do gulp serve. Is it due to tslint ?

Could you please make a sandbox with a basic set up so I can test it?

Regards

**Comment by mhammadsiddiqui** [comment]
> > I am having th e same issue while using,
> > "jodit-react": "^1.1.1",
> > "typescript": "^4.4.2"
> > and "skipLibCheck": true, option in my tsconfig.json.
> > I can not do gulp serve. Is it due to tslint ?
> 
> Could you please make a sandbox with a basic set up so I can test it?
> 
> Regards

There is a compiling error;
![Capture](https://user-images.githubusercontent.com/42937945/132240804-5f77c023-a667-4d3b-87e0-ebb3fcd740a8.PNG)

**Comment by Remyrd77** [comment]
> > > I am having th e same issue while using,
> > > "jodit-react": "^1.1.1",
> > > "typescript": "^4.4.2"
> > > and "skipLibCheck": true, option in my tsconfig.json.
> > > I can not do gulp serve. Is it due to tslint ?
> > 
> > 
> > Could you please make a sandbox with a basic set up so I can test it?
> > Regards
> 
> There is a compiling error; ![Capture](https://user-images.githubusercontent.com/42937945/132240804-5f77c023-a667-4d3b-87e0-ebb3fcd740a8.PNG)

Were you able to resolve this error ?
